app/controllers/api/v1/invoices_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Api::V1::InvoicesController < ApplicationController
+  def index
+    if params[:merchant_id]
+      merchant = Merchant.find_by(id: params[:merchant_id])
+    if merchant.nil?
+      render json: { error: "Merchant not found" }, status: :not_found
+    return
+  end
```

Comment:
formatting - tab this over

Reply:
Do we want conditional logic in the controller? or move it to the model?